recipes/boost/meta.yaml

```diff
@@ -0,0 +1,91 @@
+{% set version = "1.60.0" %}
+{% set filename = "boost_%s.tar.bz2" % version.replace(".", "_") %}
+
+package:
+  name: boost
+  version: {{ version }}
+
+source:
+  fn:  {{ filename }}
+  url: http://sourceforge.net/projects/boost/files/boost/{{ version }}/{{ filename }}
+  md5: 65a840e1a0b13a558ff19eeb2c4f0cbe
+
+build:
+  features:
+    - vc9               # [win and py27]
+    - vc10              # [win and py34]
+    - vc14              # [win and py35]
+
+requirements:
+  build:
+    - gcc               # [linux]
+    - python
+    - icu               # [unix]
+    - bzip2             # [unix]
+    - zlib
+
+  run:
+    - libgcc            # [linux]
+    - python
+    - icu               # [unix]
+    - zlib
+
+test:
+  commands:
+    # Verify static libraries.
+    - test -f $PREFIX/lib/libboost_chrono.a                   # [unix]
+    - test -f $PREFIX/lib/libboost_exception.a                # [unix]
+    - test -f $PREFIX/lib/libboost_system.a                   # [unix]
+    - test -f $PREFIX/lib/libboost_test_exec_monitor.a        # [unix]
+    - test -f $PREFIX/lib/libboost_timer.a                    # [unix]
+
+    # Verify dynamic libraries.
+    {% set boost_libs = [
+            "atomic",
+            "chrono",
+            "container",
+            "context",
+            "coroutine",
+            "date_time",
+            "filesystem",
+            "graph",
+            "iostreams",
+            "locale",
+            "log",
+            "log_setup",
+            "math_c99",
+            "math_c99f",
+            "math_c99l",
+            "math_tr1",
+            "math_tr1f",
+            "math_tr1l",
+            "prg_exec_monitor",
+            "program_options",
+            "python",
+            "random",
+            "regex",
+            "serialization",
+            "signals",
+            "system",
+            "thread",
+            "timer",
+            "type_erasure",
+            "unit_test_framework",
+            "wave",
+            "wserialization"
+    ] %}
+    {% for each_boost_lib in boost_libs %}
+    - test -f $PREFIX/lib/libboost_{{ each_boost_lib }}.dylib  # [osx]
+    - test -f $PREFIX/lib/libboost_{{ each_boost_lib }}.so     # [linux]
+    {% endfor %}
```

Comment:
Now has a nice clean jinja template based implementation. Still quite long as we have to explicitly list all the library names. However, with this strategy we don't need to explicitly list every library for every system. Instead we have library names for all systems and then system naming conventions. Tried it out on my Mac and it worked. Will now add Windows.